processor/runtime/src/main/java/com/redhat/service/smartevents/processor/actions/source/SourceActionInvokerBuilder.java

```diff
@@ -0,0 +1,29 @@
+package com.redhat.service.smartevents.processor.actions.source;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.redhat.service.smartevents.infra.auth.AbstractOidcClient;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.processor.actions.AbstractWebClientInvokerBuilder;
+import com.redhat.service.smartevents.processor.actions.ActionInvoker;
+
+@ApplicationScoped
+public class SourceActionInvokerBuilder extends AbstractWebClientInvokerBuilder implements SourceAction {
+
+    @Override
+    public ActionInvoker build(ProcessorDTO processor, Action action) {
+        String endpoint = getOrThrowException(action, ENDPOINT_PARAM);
+        String cloudEventType = getOrThrowException(action, CLOUD_EVENT_TYPE_PARAM);
+        AbstractOidcClient abstractOidcClient = getOidcClient();
+        return new SourceActionInvoker(endpoint, cloudEventType, webClient, abstractOidcClient);
+    }
+
+    private static String getOrThrowException(Action action, String parameterName) {
+        if (action.getParameters() == null || !action.getParameters().containsKey(parameterName)) {
+            throw new GatewayProviderException(String.format("Missing %s parameter from action", parameterName));
+        }
```

Comment:
Is this validation also performed during Action validation?

Reply:
There's no action validation phase for `SourceAction` because it's an internal action used only by source processors. However, the existence of those parameters should be guaranteed by the unit tests of the source resolvers.

Reply:
With the newest refactoring `SourceAction` is not needed anymore and standard `WebhookAction` can be used instead. Thus, I removed all this code.